bluemira/codes/_freecadapi.py

```diff
@@ -651,6 +651,55 @@ def dist_to_shape(shape1, shape2):
     return dist, vectors
 
 
+def slice_shape(shape: apiShape, plane_origin: Iterable, plane_axis: Iterable):
+    """
+    Slice a shape along a given plane
+
+    TODO improve face-sold-shell interface
+
+    Parameters
+    ----------
+    shape: apiShape
+        shape to slice
+    plane_origin: Iterable
+        normal plane origin
+    plane_axis: Iterable
+        normal plane axis
+
+    Notes
+    -----
+    Degenerate cases such as tangets to solid or faces do not return intersections
+    if the shape and plane are acting at the Placement base.
+    Further investigation needed.
+
+    """
+    if isinstance(shape, apiWire):
+        return _slice_wire(shape, plane_axis, plane_origin)
+    else:
+        bluemira_warn("The output structure of this function may not be as expected")
+        shift = np.dot(np.array(plane_origin), np.array(plane_axis))
+        return _slice_solid(shape, plane_axis, shift)
+
+
+def _slice_wire(wire, normal_plane, shift, *, BIG_NUMBER=1e5):
```

Comment:
I'd quite like to align the interface here with other bluemira operations, but I see that the FreeCAD interface is actually different for slice, and this aligns with FreeCAD... perhaps ok for now.

Reply:
This should only really be called in the above `slice_shape` functionwhich should only really be called from tools. What interface changes are thinking of?

Reply:
I was thinking along the lines of `rotate_shape` and so on where there's a `base` and `direction` parameter... but thinking about it a bit more that doesn't really make much sense since the nature of this slice operation is different from those transformative operations. No changes needed here.